Astro CLI release notes 1.9 (#1735)
astronomer/docs: astro/cli/release-notes.md

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.9.0

Release date: January 12, 2023

### Programmatically manage Deployments with the Astro CLI

Astro CLI version 1.9 includes two new commands that make it possible to programmatically manage Deployments: `astro deployment inspect --template` and `astro deployment create/update --deployment-file`.

For more information about programmatically creating and updating Deployments, see [Manage Deployments as code](manage-deployments-as-code.md).

### New `--dag-file` flag for `astro run`

By default, `astro run` parses all the DAGs in your `dags` directory even if you are only running one DAG. You can now use the `--dag-file` flag to limit the CLI to parsing and running only the specified DAG file. 

### Additional improvments

- When you run Airflow locally, you no longer need to enter credentials to log in to the Airflow UI.
- The Airflow UI now includes **Configurations** in the **Admin** menu. Use this page to see all current configurations for your Airflow environment. 

> - The Astro CLI now reminds you when a new version of the CLI is available. To turn this feature off, run 'astro config set -g upgrade_message false'.

Reviewer: ```suggestion
- The Astro CLI now reminds you when a new version of the CLI is available. To turn this feature off, run `astro config set -g upgrade_message false`.
```

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.9.0

Release date: January 10, 2022

### New `--template` Flag for `deployment inspect` Command

Create template files based on already existing deployments with the `--template` flag. This template deployment file will have an empty name field ,  empty description field, and will not contain the deployments metadata. These template files can be used to create new deployments with the same configuration as the original.

> ### Create and Update Deployments with Deployment Files

Reviewer: ```suggestion
### Create and update Deployments using Deployment files
```

> Users can now use the `--deployment-file` flag with the `deployment create` and `update`. These new features along with the inspect command allow users to manage their Deployments as Code. Read more about Deployments as code in our documentation.

Reviewer: ```suggestion
You can now use the `--deployment-file` flag with `astro deployment create` and `astro deployment update`. Use these flags with `astro deployment inspect` to manage your Deployments as YAML configuration files. 
```

> ### New `--template` Flag for `deployment inspect` Command
> 
> Create template files based on already existing deployments with the `--template` flag. This template deployment file will have an empty name field ,  empty description field, and will not contain the deployments metadata. These template files can be used to create new deployments with the same configuration as the original.
> 
> ### Create and Update Deployments with Deployment Files
> 
> Users can now use the `--deployment-file` flag with the `deployment create` and `update`. These new features along with the inspect command allow users to manage their Deployments as Code. Read more about Deployments as code in our documentation.

Reviewer: ```suggestion
### Programmatically manage Deployments with the Astro CLI

Astro CLI version 1.9 includes two new commands that make it possible to programmatically manage Deployments: `astro deployment inspect --template` and `astro deployment create/update --deployment-file`.

`astro deployment inspect --template` creates a template file based on the selected Deployment. A template file is YAML configuration file that includes all information about a Deployment except for its name, description field, and metadata. Template files allow you to programmatically create new Deployments outside of the Cloud UI.

After you specify a new name and description in the template file, you can apply the configuration to a new or existing Deployment with `astro deployment create/update --deployment-file`. 

For more information about programmatically creating and updating Deployments, see [Manage Deployments as code](manage-deployments-as-code.md).
```

Reviewer: Alternative if we want to combine notes

Author: I like it

### New `—dag-file` Flag for `astro run` Command

> By default The ‘astro run’ command parses all the DAGs in your dag folder, even if you only want to run 1 DAG. Now you can use the `--dag-file` flag to specify a DAG file. With this flag the command will only parse and run the DAG and file you specify.

Reviewer: ```suggestion
By default, `astro run` parses all the DAGs in your `dags` directory even if you are only running one DAG. You can now use the `--dag-file` flag to limit the CLI to parsing and running only the specified DAG file. 
```

### Additional improvments

> - You will no longer need to enter credentials in locally running airflow environments. The credentials will filled in for you.

Reviewer: ```suggestion
- When you run Airflow locally, you no longer need to enter credentials to log in to the Airflow UI.
```

> - The configurations page in the Admin menu is not available to you to view in the local Webserver UI.

Reviewer: ```suggestion
- The Airflow UI now includes **Configurations** in the **Admin** menu. Use this page to see all current configurations for your Airflow environment. 
```
I assume you meant that this page is "Now" available?

> - The Astro CLI will now tell you when a new version of the Astro CLI is available. Run 'astro config set -g upgrade_message false' to turn this feature off.

Reviewer: ```suggestion
- The Astro CLI now reminds you when a new version of the CLI is available. To turn this feature off, run 'astro config set -g upgrade_message false'.
```

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.9.0

Release date: January 12, 2023

### Programmatically manage Deployments with the Astro CLI

> Astro CLI version 1.9 includes two new commands that make it possible to programmatically manage Deployments: `astro deployment inspect --template` and `astro deployment create/update --deployment-file`.

Reviewer: I made some minor wording changes but overall LGTM

`astro deployment inspect --template` creates a template file based on the selected Deployment. A template file is YAML configuration file that includes all information about a Deployment except for its name, description field, and metadata. Template files allow you to programmatically create new Deployments outside of the Cloud UI.

After you specify a new name and description in the template file, you can apply the configuration to a new or existing Deployment with `astro deployment create/update --deployment-file`. 

For more information about programmatically creating and updating Deployments, see [Manage Deployments as code](manage-deployments-as-code.md).

### New `--dag-file` flag for `astro run`

> By default, `astro run` parses all the DAGs in your `dags` directory even if you are only running one DAG. You can now use the `--dag-file` flag to limit the CLI to parsing and running only the specified DAG file. 

Reviewer 2: ```suggestion
By default, `astro run` parses all of the DAGs in your `dags` directory even if you are only running one DAG.

With Astro CLI 1.9, you can now use the `--dag-file` flag such that the CLI parses and runs only the specified DAG file. This makes  the command quicker and allows you to avoid seeing errors from other DAGs if you don't need to see them.
```

Some sort of value statement here was missing?

### Additional improvments

> - When you run Airflow locally, you no longer need to enter credentials to log in to the Airflow UI.

Reviewer 2: @jwitz Did we make sure to update instructions in other parts of our docs where we outline having to enter a username and password?

Reviewer: Because it's such a minor point of friction, I'm actually inclined to keep the admin credentials for now. Would rather have 1.9 users see a step that doesn't apply when they're already logged into Airflow vs. an old CLI user having to search around for how to get access to the UI. Thoughts?

> - The Airflow UI now includes **Configurations** in the **Admin** menu. Use this page to see all current configurations for your Airflow environment. 

Reviewer 2: ```suggestion
- Local Airflow environments running with the Astro CLI now include access to the **Configurations** page in the **Admin** menu of the Airflow UI. Use this page to see your environment's current configuration.
```

> - The Astro CLI now reminds you when a new version of the CLI is available. To turn this feature off, run 'astro config set -g upgrade_message false'.

Reviewer 2: We missing code ticks here?

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.9.0

Release date: January 12, 2023

> ### Programmatically manage Deployments with the Astro CLI

Reviewer 2: ```suggestion
### Manage Astro Deployments as code
```

> Astro CLI version 1.9 includes two new commands that make it possible to programmatically create and update Deployments:

Reviewer 2: ```suggestion
Astro CLI version 1.9 includes three new commands that make it possible to programmatically create and update Deployments:
```

- `astro deployment inspect --template`: Create a template file in YAML for an existing Deployment. This template file includes all information about the Deployment in its current state, including worker queue configurations, environment variables, and Astro Runtime version.
- `astro deployment create --deployment-file`: Create a new Deployment with the configurations specified in a template file.

> - `astro deployment update --deployment-file`: Programmatically update an existing Deployment based on the values in a template file.

Reviewer 2: ```suggestion
- `astro deployment update --deployment-file`: Update an existing Deployment based on the values in a deployment file.
```

> Template files are a way for your team to define Astro Deployments as code. If your team regularly creates and deletes Deployments for testing, for example, you can use template files to do that more quickly without manually copying configurations in the Cloud UI.

Reviewer 3: ```suggestion
You can use template files to define Astro Deployments as code. For example, if your team regularly creates and deletes Deployments for testing, you can use template files to avoid manually copying configurations in the Cloud UI.
```

> For more information about programmatically creating and updating Deployments, see [Manage Deployments as code](manage-deployments-as-code.md).

Reviewer 2: ```suggestion
```

### New `--dag-file` flag for `astro run`

By default, `astro run` parses all of the DAGs in your `dags` directory even if you are only running one DAG.

> With Astro CLI 1.9, you can now use the `--dag-file` flag such that the CLI parses and runs only the specified DAG file. This makes  the command quicker and allows you to avoid seeing errors from other DAGs if you don't need to see them.

Reviewer 3: ```suggestion
In Astro CLI 1.9, you can use the `--dag-file` flag to parse and run a specific DAG file. Specifying a specific DAG file speeds command execution and reduces the error messages returned.
```

### Additional improvments

- When you run Airflow locally, you no longer need to enter credentials to log in to the Airflow UI.

> - Local Airflow environments running with the Astro CLI now include access to the **Configurations** page in the **Admin** menu of the Airflow UI. Use this page to see your environment's current configuration.

Reviewer 3: ```suggestion
- Local Airflow environments running with the Astro CLI now include access to the Airflow UI **Configurations** page (**Admin** > **Configurations**). This page shows your the configuration for your environment.
```

> - The Astro CLI now reminds you when a new version of the CLI is available. To turn this feature off, run `astro config set -g upgrade_message false`.

Reviewer 3: ```suggestion
- The Astro CLI now reminds you when a new version of the Astro CLI is available. To turn this feature off, run `astro config set -g upgrade_message false`.
```
